### PR TITLE
Changed structure for version's drop-down menu & styled the element properly #10088

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

PR #10079 introduced some changes which add an extra `ul` element. So far existing `li.dropdown-item` has been a simple `<li>` with text. After the changes, it became `<li>` containing an extra `<ul>` which shows nested versions. It also has added some spacing for nested `<li>` elements.

This PR put the newly created `<ul>` element into `<section>` and achieved nesting elements by extending `NavLink` component. It seems to be much more clear approach.

[skip changelog]

### How has this been tested?

Tested locally, using cases from the issue.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #10014
2. #10088

### CODE REVIEW

**Review by budnix — COMMENTED** [review]
Comparing the dropdown list with the latest docs version, it seems that they differ. Is that is intentional?

localhost:
image

Docs 12.2 and lower
image

**Comment by wszymanski** [comment]
> Comparing the dropdown list with the latest docs version, it seems that they differ. Is that is intentional?

Thank you, it wasn't intentional. Changed within https://github.com/handsontable/handsontable/pull/10091/commits/07f4bc8dd311c84cda93f8fbc9fae24787d437ad according to new menu structure.

**Review by budnix — DISMISSED** [review]

**Comment by wszymanski** [comment]
@budnix found yet another difference (see below)

image

I've found yet another problem, related to no external link icon, which could be affected by the change. I'm checking it.
 
diff

**Comment by wszymanski** [comment]
@budnix I've rethought part of the solution introduced within #10079 and I think that some changes are needed.

The above PR changed code for `DropdownLink.vue` component. It is used by `VersionsDropdown.vue`, `FrameworksDropdown.vue` and `ExternalNavLinks.vue` components. Removing used internally `NavLink` in their code caused some bugs.

To fulfil all needs I can:

1. Extend the `NavLink` (current code)

or

2. Create new `NavLinkSection` component which contain already added code. 

Please let me know what you think about it. The latest commit reverts `docs/.vuepress/theme/components/DropdownLink.vue` file, change a little bit names, extend the `NavLink.vue` and doing DOM elements from drop-down menu looking more accurate (introduced `<section>` and `<p>` elements, not creating an `<ul>` for single link).

<table>
<tr>
	<td>Was
	<td>Is
<tr>
	<td>
	<td>

</table>



I'll change PR title and description if it looks good for you.

**Review by budnix — DISMISSED** [review]
Looks good 👌

File: docs/.vuepress/theme/components/NavLink.vue [hunk 1]
```diff
@@ -16,7 +16,22 @@
     :rel="rel"
     @focusout="focusoutAction"
   >
-    {{ item.text }}
+    <section v-if="item.subTexts?.length">
```
Comment (budnix) [inline docs/.vuepress/theme/components/NavLink.vue:19]:
```suggestion
    <section v-if="item.subTexts && item.subTexts.length">
```
With optional chaining, I've got errors in watch and build modes. I tried with fresh dependencies, and the results were the same.

```
client?cae4:159 ./.vuepress/theme/components/NavLink.vue?vue&type=template&id=0af88b9c& (./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/@vuepress/core/node_modules/.cache/vuepress","cacheIdentifier":"febc8548-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./.vuepress/theme/components/NavLink.vue?vue&type=template&id=0af88b9c&)
Module build failed (from ./node_modules/vue-loader/lib/loaders/templateLoader.js):
SyntaxError: Unexpected token (1:388)
    at Parser.pp$4.raise (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:2757:13)
    at Parser.pp.unexpected (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:647:8)
    at Parser.pp$3.parseExprAtom (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:2196:10)
    at Parser.<anonymous> (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:6003:24)
    at Parser.parseExprAtom (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:6129:31)
    at Parser.pp$3.parseExprSubscripts (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:2047:19)
    at Parser.pp$3.parseMaybeUnary (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:2024:17)
    at Parser.pp$3.parseExprOps (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:1966:19)
    at Parser.pp$3.parseMaybeConditional (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:1949:19)
    at Parser.pp$3.parseMaybeAssign (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:1925:19)
    at Parser.pp$3.parseMaybeConditional (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:1954:28)
    at Parser.pp$3.parseMaybeAssign (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:1925:19)
    at Parser.pp$3.parseParenAndDistinguishExpression (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:2238:30)
    at Parser.pp$3.parseExprAtom (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:2163:41)
    at Parser.<anonymous> (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:6003:24)
    at Parser.parseExprAtom (/Users/budnix/Documents/Projects/handsontable-develop/docs/node_modules/vue-template-es2015-compiler/buble.js:6129:31)
```

Comment (wszymanski) [reply docs/.vuepress/theme/components/NavLink.vue:19]: It doesn't apply to my local env and staging, but I'll change it.

**Review by budnix — APPROVED** [review]

**Pull request merged**